lib/improver/blending/spatial_weights.py

```diff
@@ -0,0 +1,275 @@
+           supplied to the plugin.
+        4. Normalises the weights along the coordinatethat will be collapsed
+           when blending is carried out using these weights.
+    """
+
+    def __init__(self, fuzzy_length=10):
+        """
+        Initialise class.
+        Keyword Args:
+            fuzzy_length (int or float):
+                The length in terms of number of grid squares, over which the
+                weights from the input data mask are smoothed.
+        """
+        self.fuzzy_length = fuzzy_length
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<SpatialWeightsForMissingData: fuzzy_length: {}>'.format(
+                      self.fuzzy_length))
+        return result
+
+    @staticmethod
+    def create_initial_weights_from_mask(cube):
+        """
+        Generate a cube with weights generated from the mask of the input cube.
+        Where the data is masked we set the weight to zero, otherwise the
+        weight is one.
+
+        Args:
+            cube (iris.cube.Cube):
+                A cube containing the data we want to collapse by doing a
+                weighted blend along a given cordinate.
+        Return:
+            masked_weights_cube (iris.cube.Cube):
+                A cube containing an initial set of weights based on the mask
+                on the input cube.
+        Rasies:
+            ValueError : If the input cube does not have a mask.
+        """
+        if np.ma.is_masked(cube.data):
+            weights_data = np.where(cube.data.mask, 0, 1).astype(np.float32)
+        else:
+            message = ("Input cube to SpatialWeightsForMissingData "
+                       "must be masked")
+            raise ValueError(message)
+        masked_weights_cube = cube.copy(data=weights_data)
+        masked_weights_cube.rename("weights")
+        return masked_weights_cube
+
+    def create_fuzzy_spatial_weights(self, masked_weights_cube):
+        """
+        Create fuzzy weights around points in the masked_weights_cube with
+        zero weight.
+
+        Args:
+            masked_weights_cube (iris.cube.Cube):
+                A cube containing an initial set of weights based on the mask
+                on the input cube.
+        Returns:
+            result (iris.cube.Cube):
+                A cube contiaing the fuzzy weights calculated based on the
+                masked_weights_cube. The dimension order may have changed from
+                the input cube as it has been sliced over x and y coordinates.
+        """
+        result = iris.cube.CubeList()
+        x_coord = masked_weights_cube.coord(axis='x').name()
+        y_coord = masked_weights_cube.coord(axis='y').name()
+        # The distance_transform_edt works on N-D cubes, so we want to make
+        # sure we only apply it to x-y slices.
+        for weights in masked_weights_cube.slices([x_coord, y_coord]):
```

Comment:
Why are we slicing in the x-y order, when y-x is the IMPROVER convention?  It would be worth explicit mention (comment or docstring) if this is necessary.